Flaky: unit test TestSDKServerAddListValue

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via 
-->


**What happened**:
```
Step #20 - "tests": --- FAIL: TestSDKServerAddListValue (31.98s)
Step #20 - "tests":     --- FAIL: TestSDKServerAddListValue/Add_multiple_values_past_capacity (1.12s)
Step #20 - "tests":         sdkserver_test.go:1477: 
Step #20 - "tests":             	Error Trace:	/go/src/agones.dev/agones/pkg/sdkserver/sdkserver_test.go:1477
Step #20 - "tests":             	Error:      	Not equal: 
Step #20 - "tests":             	            	expected: v1.ListStatus{Capacity:10, Values:[]string{"one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"}}
Step #20 - "tests":             	            	actual  : v1.ListStatus{Capacity:10, Values:[]string{"one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}}
Step #20 - "tests":             	            	
Step #20 - "tests":             	            	Diff:
Step #20 - "tests":             	            	--- Expected
Step #20 - "tests":             	            	+++ Actual
Step #20 - "tests":             	            	@@ -2,3 +2,3 @@
Step #20 - "tests":             	            	  Capacity: (int64) 10,
Step #20 - "tests":             	            	- Values: ([]string) (len=10) {
Step #20 - "tests":             	            	+ Values: ([]string) (len=9) {
Step #20 - "tests":             	            	   (string) (len=3) "one",
Step #20 - "tests":             	            	@@ -11,4 +11,3 @@
Step #20 - "tests":             	            	   (string) (len=5) "eight",
Step #20 - "tests":             	            	-  (string) (len=4) "nine",
Step #20 - "tests":             	            	-  (string) (len=3) "ten"
Step #20 - "tests":             	            	+  (string) (len=4) "nine"
Step #20 - "tests":             	            	  }
Step #20 - "tests":             	Test:       	TestSDKServerAddListValue/Add_multiple_values_past_capacity
```

**What you expected to happen**:
Tests should pass.

**How to reproduce it (as minimally and precisely as possible)**:
Not entirely sure. Seems to mostly happen in CI, have yet to try and reproduce locally.

**Anything else we need to know?**:

**Environment**:
- Agones version: dev
- Kubernetes version (use `kubectl version`): varies
- Cloud provider or hardware configuration: GKE
- Install method (yaml/helm): helm
- Troubleshooting guide log(s): see above what happened section
- Others: N/A

--- a/pkg/sdkserver/sdkserver_test.go
+++ b/pkg/sdkserver/sdkserver_test.go
@@ -1437,16 +1437,8 @@ func TestSDKServerAddListValue(t *testing.T) {
 			require.NoError(t, err)
 			assert.NoError(t, sc.WaitForConnection(ctx))
 			sc.informerFactory.Start(ctx.Done())
-			assert.True(t, cache.WaitForCacheSync(ctx.Done(), sc.gameServerSynced))
-
-			wg := sync.WaitGroup{}
-			wg.Add(1)
-
-			go func() {
-				err = sc.Run(ctx)
-				assert.NoError(t, err)
-				wg.Done()
-			}()
+			require.True(t, cache.WaitForCacheSync(ctx.Done(), sc.gameServerSynced))
+			sc.gsWaitForSync.Done()
 
 			// check initial value comes through
 			require.Eventually(t, func() bool {
@@ -1469,6 +1461,14 @@ func TestSDKServerAddListValue(t *testing.T) {
 			assert.Equal(t, testCase.want.Values, got.Values)
 			assert.Equal(t, testCase.want.Capacity, got.Capacity)
 
+			// start workerqueue processing at this point, so there is no chance of processing the above updates
+			// earlier.
+			sc.gsWaitForSync.Add(1)
+			go func() {
+				err = sc.Run(ctx)
+				assert.NoError(t, err)
+			}()
+
 			// on an update, confirm that the update hits the K8s api
 			if testCase.updated {
 				select {
@@ -1487,7 +1487,6 @@ func TestSDKServerAddListValue(t *testing.T) {
 			assert.Equal(t, testCase.expectedUpdatesQueueLen, glu)
 
 			cancel()
-			wg.Wait()
 		})
 	}
 }
@@ -1586,16 +1585,8 @@ func TestSDKServerRemoveListValue(t *testing.T) {
 			require.NoError(t, err)
 			assert.NoError(t, sc.WaitForConnection(ctx))
 			sc.informerFactory.Start(ctx.Done())
-			assert.True(t, cache.WaitForCacheSync(ctx.Done(), sc.gameServerSynced))
-
-			wg := sync.WaitGroup{}
-			wg.Add(1)
-
-			go func() {
-				err = sc.Run(ctx)
-				assert.NoError(t, err)
-				wg.Done()
-			}()
+			require.True(t, cache.WaitForCacheSync(ctx.Done(), sc.gameServerSynced))
+			sc.gsWaitForSync.Done()
 
 			// check initial value comes through
 			require.Eventually(t, func() bool {
@@ -1618,6 +1609,14 @@ func TestSDKServerRemoveListValue(t *testing.T) {
 			assert.Equal(t, testCase.want.Values, got.Values)
 			assert.Equal(t, testCase.want.Capacity, got.Capacity)
 
+			// start workerqueue processing at this point, so there is no chance of processing the above updates
+			// earlier.
+			sc.gsWaitForSync.Add(1)
+			go func() {
+				err = sc.Run(ctx)
+				assert.NoError(t, err)
+			}()
+
 			// on an update, confirm that the update hits the K8s api
 			if testCase.updated {
 				select {
@@ -1636,7 +1635,6 @@ func TestSDKServerRemoveListValue(t *testing.T) {
 			assert.Equal(t, testCase.expectedUpdatesQueueLen, glu)
 
 			cancel()
-			wg.Wait()
 		})
 	}
 }

